### ISSUE TITLE
InvalidOperationException in MSBuild when building in VS

### ISSUE THREAD

**Issue by jdrobison** [dotnet/msbuild#7781]
### Issue Description
I have a filtered solution containing three projects that builds fine from the command line and until recently would also build fine in VS.  After installing the latest Int Preview build (32630.467.main), all three projects fail to build.  The binlog produced by [Project System Tools 2002](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProjectSystemTools2022) shows several instances InvalidOperationExceptions from MSBuild, for example:

```
UNHANDLED EXCEPTIONS FROM PROCESS 27748:
=====================
7/1/2022 2:01:05 AM
System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at Microsoft.Build.Collections.RetrievableEntryHashSet`1.Enumerator.MoveNext()
   at Microsoft.Build.BackEnd.Logging.ProjectLoggingContext.<FilterEnvironmentDerivedProperties>d__3.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at Microsoft.Build.Framework.ProjectStartedEventArgs.WriteToStream(BinaryWriter writer)
   at Microsoft.Build.Shared.LogMessagePacketBase.WriteToStream(ITranslator translator)
   at Microsoft.Build.BackEnd.NodeEndpointOutOfProcBase.RunReadLoop(Stream localReadPipe, Stream localWritePipe, ConcurrentQueue`1 localPacketQueue, AutoResetEvent localPacketAvailable, AutoResetEvent localTerminatePacketPump)
===================
```

One such binlog is attached.

### Versions & Configurations
```
D:\vs.retail\src\env\shell\UIInternal> msbuild -version
MSBuild version 17.3.0-preview-22329-01+77c72dd0f for .NET Framework
17.3.0.32901
```

[UIInternal_Debug_AnyCPU_Build_2022-07-01T10_02_23.8126443-07_00.zip](https://github.com/dotnet/msbuild/files/9030827/UIInternal_Debug_AnyCPU_Build_2022-07-01T10_02_23.8126443-07_00.zip)

**Comment by rainersigwald** [comment]
This should be fixed in the next int preview build. The workaround is to disable the Project System Tools extension.

**Comment by rainersigwald** [comment]
Fixed by #7766

**Comment by jdrobison** [comment]
Disabling the extension worked great.  Thanks for the workaround.
